mersenne-31/src/x86_64_avx2/poseidon2.rs

```diff
@@ -1,13 +1,26 @@
+use alloc::vec::Vec;
 use core::arch::x86_64::{self, __m256i};
+
 use p3_poseidon2::{
     mds_light_permutation, ExternalLayer, InternalLayer, MDSMat4, Poseidon2ExternalPackedConstants,
     Poseidon2InternalPackedConstants,
 };
 
-use crate::{
-    exp5, Mersenne31, PackedMersenne31AVX2, Poseidon2ExternalLayerMersenne31,
-    Poseidon2InternalLayerMersenne31, P, P_AVX2,
-};
+use crate::{exp5, Mersenne31, PackedMersenne31AVX2, P, P_AVX2};
+
+#[derive(Debug, Clone, Default)]
+pub struct Poseidon2InternalLayerMersenne31 {
+    pub(crate) internal_constants: Vec<Mersenne31>,
```

Comment:
Do we need to keep the input format, seems like it could be forgotten?

Reply:
Hmm, I think we need to keep it if we want to be able to run Poseidon2 on lists of field elements, e.g. [F; 16] and not just lists of packed field elements e.g. [PF; 16].